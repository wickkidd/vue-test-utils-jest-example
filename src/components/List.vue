<template>
  <div>
    <ul>
      <li :key="item" v-for="item in items">
        {{ item }}
      </li>
    </ul>

    <AddItemButton @addItem="$emit('addItem', $event)"></AddItemButton>
  </div>
</template>

<script>
  import AddItemButton from '@/components/AddItemButton.vue'

  export default {
    name: 'List',
    components: {AddItemButton},
    props: [
      'items'
    ]
  }
</script>
