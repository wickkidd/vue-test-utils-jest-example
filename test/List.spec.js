import { shallowMount } from '@vue/test-utils'
import List from '@/components/List.vue'

describe('List', () => {
  const items = ['item 1', 'item 2']
  const wrapper = shallowMount(List, {
    propsData: { items }
  })

  it('renders li for each item in props.items', () => {
    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
