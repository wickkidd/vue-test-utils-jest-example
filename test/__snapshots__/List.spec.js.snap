// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot 1`] = `
<div>
  <ul>
    <li>
      item 1
    </li>
    <li>
      item 2
    </li>
  </ul>
  <additembutton-stub></additembutton-stub>
</div>
`;
